src/app/config/constants/constants.ts

```diff
@@ -1,26 +1,35 @@
 import { HttpClient } from '@angular/common/http';
 import { Injectable } from '@angular/core';
+import { BehaviorSubject } from 'rxjs';
 
 import apiURL from 'src/app/config/constants/apiURL';
+
 interface externalNameResponseFormat {
-  externalName: String;
+  externalName: string;
 };
 
-@Injectable({
-  providedIn: 'root',
-})
+@Injectable({ providedIn: 'root' })
 export class DoubtfireConstants {
-  constructor(private httpClient: HttpClient) {
-    httpClient.get(`${this.apiURL}/settings`).toPromise().then((response: externalNameResponseFormat) => {
-      this.externalName = response.externalName;
-    })
+  constructor(private http: HttpClient) {
+    this.loadExternalName();
   }
   public mainContributors: string[] = [
     'macite',              // Andrew Cain
     'alexcu',              // Alex Cummaudo
     'jakerenzella'         // Jake Renzella
   ];
-  public apiURL = apiURL;
-  public externalName: String = "Loading...";
-}
 
+  public loaded: boolean = false;
+  public apiURL: string = apiURL;
+
+  public getExternalName = new BehaviorSubject<string>('Loading...');
```

Comment:
Nice!

Reply:
Can we rename? Maybe just `ExternalName` or `ExternalNameSubject`?

Reply:
👍

Reply:
This is where we needed the rename... as it is a subject, I think we remove "get". So it would be "ExternalName.next" and "ExternalName.subscribe" etc.